### PR TITLE
#1830 hide message quote

### LINKED ISSUE #1830: Add ability to hide quote text in thread message
Currently each thread message shows quote of all previous messages in the end of text.
It makes each message cell very long in thread view and more scrolling is needed for finding specific message.

I propose to add `Show history` button instead of thread quote text, so all previous messages texts will be shown only after button tap.

### PR DESCRIPTION
**Pull request by sosnovsky**
This PR implements automatic hiding of quote text

close #1830

----------------------------------

**Tests** _(delete all except exactly one)_:
- Tests added or updated

--------------------------------

### To be filled by reviewers

I have reviewed that this PR... _(tick whichever items you personally focused on during this review)_:
- [x] addresses the issue it closes (if any)
- [x] code is readable and understandable
- [x] is accompanied with tests, or tests are not needed
- [x] is free of vulnerabilities
- [x] is documented clearly and usefully, or doesn't need documentation

### CODE REVIEW

File: appium/tests/helpers/ElementHelper.ts [hunk 1]
```diff
@@ -56,6 +56,19 @@ class ElementHelper {
     await element.click();
   }
 
+  static deleteText = async (element: WebdriverIO.Element) => {
```
Comment (ioanmo226) [inline appium/tests/helpers/ElementHelper.ts:59]: I was confused about `deleteText` name. I thought this function also deletes text for buttons.
Maybe need to rename it to `deleteInputText` or `clearInput` or something like that?

**Review by ioanmo226 — COMMENTED** [review]
👍 
Looks great. Just a few questions/suggestions.

File: appium/tests/helpers/ElementHelper.ts (hunk 1 above)
Comment (sosnovsky) [reply appium/tests/helpers/ElementHelper.ts:59]: `clearInput` sounds good 👍

File: FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift [hunk 2]
```diff
@@ -65,19 +76,75 @@ struct ProcessedMessage {
...
+                } else {
+                    lines.append(lastLine)
+                }
+                break
+            }
+        }
+        let hasQuote = quoteLines.contains { !$0.isEmpty }
+        let quote = hasQuote ? quoteLines.joined(separator: "\n") : nil
+        let message = lines.joined(separator: "\n").trimmingCharacters(in: .whitespacesAndNewlines)
+        return (message, quote)
```
[the remark this answers is not in the document]
Comment (sosnovsky) [reply FlowCrypt/Functionality/Mail Provider/Message Provider/ProcessedMessage.swift:135]: Yes, I checked browser logic before implementing iOS behavior, and also tried some other methods for parsing quotes (like trying to find start index of message quote).
Lines looping looks like the most stable solution (but not the most elegant :))

Regex for such parsing will be quite complicated (as some lines can have spaces before `>`, quote can include empty lines, `On ... wrote:` line can be not present etc)

**Review by ioanmo226 — APPROVED** [review]
👍

**Pull request merged**